File: server/src/infra/inmemory/inmemory.go
package inmemory

import (
	"errors"
	"server/src/core/entity"
)

type InMemoryRepository struct{}

var accounts = []entity.Account{{Address: "abc", Balance: 15}, {Address: "def", Balance: 20}}

func (r InMemoryRepository) Get(address string) (entity.Account, error) {
	for _, account := range accounts {
		if account.Address == address {
			return account, nil
		}
	}
	return entity.Account{}, errors.New("account not found")
}

func (r InMemoryRepository) GetAll() []entity.Account {
	return accounts
}

File: server-ts/src/infra/inmemory.ts
class InMemoryRepository implements Repository<Account> {
  static accounts: Account[] = [
    { address: "abc", balance: 22 },
    { address: "def", balance: 15 },
  ];

  get(address: string): Promise<Account> {
    const account = InMemoryRepository.accounts.find((elem) => elem.address === address);
    return new Promise((resolve) => resolve(account));
  }

  getAll(): Promise<Account[]> {
    const accounts = InMemoryRepository.accounts;
    return new Promise((resolve) => resolve(accounts));
  }
}

export default InMemoryRepository;
